client/app/lib/components/core/fields/TextField.tsx

```diff
@@ -9,16 +9,30 @@ import {
   InputAdornment,
   TextField as MuiTextField,
   IconButton,
+  Typography,
 } from '@mui/material';
 import { VisibilityOff, Visibility } from '@mui/icons-material';
+import messagesTranslations from 'lib/translations/messages';
+import useTranslation from 'lib/hooks/useTranslation';
 
 type TextFieldProps = ComponentProps<typeof MuiTextField> & {
   trims?: boolean;
+  onChangePasswordVisibility?: (visibility: boolean) => void;
```

Comment:
Separating `PasswordTextField` was surprisingly not trivial. I imagined a future where we might have so many different descendants of `TextField` that we can no longer deal with simple if-clauses.

I tried to make `core/fields/TextField` that can have dynamic props based on the `type` (i.e., if `number`, extend `ComponentProps<typeof NumberTextField>`), but the typing was so complicated and `createElement` for some reasons keep rejecting my mapped `*TextField`s.

For now, we are stuck with `FormTextField` that has dynamic props, but internally still uses if-clauses to resolve the concrete `*TextField` elements; 50% beauty achieved(?)

>**Note** The type casting of `elementProps` in `FormTextField` is necessary because for some reasons, TypeScript keeps complaining that the `variant` prop is incompatible, despite it not being extracted/modified anywhere. I think this is one of TypeScripts edge cases; I have seen similar cases in my friends' projects.

I think the best solution is to not deal with this at all. We can continue to have loose `*TextField`s that compose `core/fields/TextField`, but instead of having `FormTextField`, we can just have a generic `FormTextField` or `FieldOf` components that works like:
```jsx
<FieldOf kind={PasswordTextField} ... /> // <-- props here is of PasswordTextField's
```
or
```jsx
// `kind` defaults to core/fields/TextField
<FormTextField kind={PasswordTextField} ... /> // <-- props here is of PasswordTextField's
```
The former is better in general because it decouples our core components from our forms, but may take some time to clear out `form/fields` and everyone that uses it. The latter is easier is faster to implement 🤞 but does not really solve the root of the problem.

Both of these solutions, however, expose the 'internality' of our core components. Our current implementation encapsulates whatever `*TextField`, which is good.

Anyway, thought it'd be nice to document this here.